sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/OpenAIClientImpl.java

```diff
@@ -1611,57 +1894,51 @@ public Response<BinaryData> getChatCompletionsWithAzureExtensionsWithResponse(St
      * </p>
      * <pre>{@code
      * {
-     *     id: String (Required)
      *     created: long (Required)
-     *     expires: Long (Optional)
-     *     result (Optional): {
-     *         created: long (Required)
-     *         data: BinaryData (Required)
-     *     }
-     *     status: String(notRunning/running/succeeded/canceled/failed) (Required)
-     *     error (Optional): {
-     *         code: String (Required)
-     *         message: String (Required)
-     *         target: String (Optional)
-     *         details (Optional): [
-     *             (recursive schema, see above)
-     *         ]
-     *         innererror (Optional): {
-     *             code: String (Optional)
-     *             innererror (Optional): (recursive schema, see innererror above)
+     *     data (Required): [
+     *          (Required){
+     *             url: String (Optional)
+     *             b64_json: String (Optional)
+     *             revised_prompt: String (Optional)
      *         }
-     *     }
+     *     ]
      * }
      * }</pre>
      * 
+     * @param deploymentOrModelName Specifies either the model deployment name (when using Azure OpenAI) or model name
+     * (when using non-Azure OpenAI) to use for this request.
      * @param imageGenerationOptions Represents the request data used to generate images.
      * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
      * @throws HttpResponseException thrown if the request is rejected by server.
      * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
      * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
      * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @return a polling status update or final response payload for an image operation along with {@link Response} on
-     * successful completion of {@link Mono}.
+     * @return the result of a successful image generation operation along with {@link Response} on successful
+     * completion of {@link Mono}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    private Mono<Response<BinaryData>> beginAzureBatchImageGenerationWithResponseAsync(
+    public Mono<Response<BinaryData>> getImageGenerationsWithResponseAsync(String deploymentOrModelName,
```

Comment:
For naming consistency, we don't need to add suffix `Async` to the method name. 
Suggestion: rename `getImageGenerationsWithResponseAsync` to `getImageGenerationsWithResponse`.

Also, Just a thought, we can shorten the name to `getImagesWithResponse` if that aligns with other languages. But I saw codegen uses `xxxImageGenerations` https://github.com/Azure/azure-sdk-for-java/pull/37739/files#diff-9f3b8c373fe094bd63c683cf0026bec4e39b75490a1a0f83d1eb43dc03fac251R1514

Maybe a naming discussion could align with the TypeSpec definition in the future.